src/ilasm/assembler.cpp

```diff
@@ -698,14 +698,19 @@ void Assembler::StartMethod(__in __nullterminated char* name, BinStr* sig, CorMe
             {
                 if(IsTdInterface(m_pCurClass->m_Attr))
                 {
-                    if(!IsMdPublic(flags)) report->error("Non-public instance method in interface\n");
-                    if((!(IsMdVirtual(flags) && IsMdAbstract(flags))))
+                    // Allow explicit method override in interfaces on C#
+                    // @DIM_TODO - Add other checks once we made a decision on which visibility flags are allowed
+                    // if(!IsMdPublic(flags)) report->error("Non-public instance method in interface\n");
+
+                    // Allow both abstract and non-abstract virtual slots (for default interface method support)
+                    if(!IsMdVirtual(flags))
                     {
-                        if(OnErrGo) report->error("Non-virtual, non-abstract instance method in interface\n");
+                        // @DIM_TODO - What about non-virtuals?
+                        if(OnErrGo) report->error("Non-virtual instance method in interface\n");
                         else
                         {
-                            report->warn("Non-virtual, non-abstract instance method in interface, set to such\n");
-                            flags = (CorMethodAttr)(flags |mdVirtual | mdAbstract);
+                            report->warn("Non-virtual instance method in interface, set to such\n");
+                            flags = (CorMethodAttr)(flags | mdVirtual);
```

Comment:
I think that in ILASM we should permit non-virtuals. This would be to allow either the feature to be implemented in the runtime, or for tests that the feature is blocked by the typeloader to be put in place. In either case, we shouldn't mystically mark the method as virtual. (However, we should consider spewing a warning)